### PR TITLE
update: save filter only when test passes

### PR DESCRIPTION
**Pull request by Oluwadaminiola**

### CODE REVIEW

File: web/ui/dashboard/src/app/private/components/create-subscription-filter/create-subscription-filter.component.html [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
+			<convoy-button size="sm" type="outline" className="mr-16px">
+				Test Filter
+				<svg width="16" height="16" class="ml-8px" class="ml-4px">
+					<use xlink:href="#test-icon"></use>
+				</svg>
+			</convoy-button>
+			<convoy-button size="sm" buttonType="button" className="min-w-[130px] min-h-[34px]" [disable]="!testResponse" (clickItem)="setSubscriptionFilter()">Save</convoy-button>
+		</div>
+	</div>
+	<form [formGroup]="subscriptionFilterForm" (ngSubmit)="testFilter()" class="bg-[#FCFCFC] rounded-8px">
```
Comment (horlah) [inline web/ui/dashboard/src/app/private/components/create-subscription-filter/create-subscription-filter.component.html:17]: Why are we using a form for this?
Isn't this adding more complexities?

File: web/ui/dashboard/src/app/private/components/create-subscription-filter/create-subscription-filter.component.html [hunk 2]
```diff
@@ -1,24 +1,24 @@
...
+			<h4>Provide your filter specifications</h4>
 		</div>
+		<div class="flex items-center">
+			<convoy-button size="sm" type="outline" className="mr-16px">
+				Test Filter
+				<svg width="16" height="16" class="ml-8px" class="ml-4px">
+					<use xlink:href="#test-icon"></use>
+				</svg>
+			</convoy-button>
+			<convoy-button size="sm" buttonType="button" className="min-w-[130px] min-h-[34px]" [disable]="!testResponse" (clickItem)="setSubscriptionFilter()">Save</convoy-button>
```
Comment (horlah) [inline web/ui/dashboard/src/app/private/components/create-subscription-filter/create-subscription-filter.component.html:14]: When you save, run the test one more time, then save. Changes might have been made that might not pass.

File: web/ui/dashboard/src/app/private/components/create-subscription-filter/create-subscription-filter.component.html [hunk 3]
```diff
@@ -1,24 +1,24 @@
...
+		</div>
+	</div>
+	<form [formGroup]="subscriptionFilterForm" (ngSubmit)="testFilter()" class="bg-[#FCFCFC] rounded-8px">
 		<div class="flex">
 			<div class="flex flex-col w-full">
 				<div class="text-14 border-b border-b-grey-20 pl-28px pt-20px pb-20px font-semibold bg-[#FCFCFC] rounded-tl-8px">Event Payload</div>
-				<convoy-monaco [editorValue]="payload" #requestEditor></convoy-monaco>
+				<convoy-monaco #requestEditor></convoy-monaco>
```
Comment (horlah) [inline web/ui/dashboard/src/app/private/components/create-subscription-filter/create-subscription-filter.component.html:21]: Why was this removed?

File: web/ui/dashboard/src/app/private/components/create-subscription-filter/create-subscription-filter.component.ts [hunk 4]
```diff
@@ -19,43 +19,28 @@ export class CreateSubscriptionFilterComponent implements OnInit {
 	@ViewChild('schemaEditor') schemaEditor!: MonacoComponent;
 	@Input('action') action: 'update' | 'create' = 'create';
 	@Input('schema') schema!: string;
-	@Input('payload') payload!: string;
 	@Output('filterSchema') filterSchema: EventEmitter<any> = new EventEmitter();
 	subscriptionFilterForm: FormGroup = this.formBuilder.group({
 		request: [null],
 		schema: [null]
 	});
+	testResponse = false;
```
Comment (horlah) [inline web/ui/dashboard/src/app/private/components/create-subscription-filter/create-subscription-filter.component.ts:27]: A better name please; `isFilterTestPassed`, `filterTestPass`?

**Review by horlah — CHANGES_REQUESTED** [review]

File: web/ui/dashboard/src/app/private/components/create-subscription-filter/create-subscription-filter.component.html (hunk 3 above)
Comment (Oluwadaminiola) [reply web/ui/dashboard/src/app/private/components/create-subscription-filter/create-subscription-filter.component.html:21]: This was taken out as the events data editor should always remain empty until a user decides to fill it, the only value stored is the filter schema

File: web/ui/dashboard/src/app/private/components/create-subscription-filter/create-subscription-filter.component.html (hunk 1 above)
Comment (Oluwadaminiola) [reply web/ui/dashboard/src/app/private/components/create-subscription-filter/create-subscription-filter.component.html:17]: I'm actually taking this out

File: web/ui/dashboard/src/app/private/components/create-subscription-filter/create-subscription-filter.component.html (hunk 3 above)
Comment (horlah) [reply web/ui/dashboard/src/app/private/components/create-subscription-filter/create-subscription-filter.component.html:21]: Yea that makes sense. The predefined value of payload here should be the payload of the latest event received or sent, so what was there was a placeholder. @Oluwadaminiola

File: web/ui/dashboard/src/app/private/components/create-subscription-filter/create-subscription-filter.component.ts [hunk 5]
```diff
@@ -19,50 +19,44 @@ export class CreateSubscriptionFilterComponent implements OnInit {
...
 
 	async testFilter() {
+		this.isFilterTestPassed = false;
 		this.subscriptionFilterForm.value.request = this.convertStringToJson(this.requestEditor.getValue());
 		this.subscriptionFilterForm.value.schema = this.convertStringToJson(this.schemaEditor.getValue());
 		try {
 			const response = await this.createSubscriptionService.testSubsriptionFilter(this.subscriptionFilterForm.value);
 			const testResponse = `The sample data was ${!response.data ? 'not' : ''} accepted by the filter`;
 			this.generalService.showNotification({ message: testResponse, style: !response.data ? 'error' : 'success' });
+			this.isFilterTestPassed = response.data;
```
Comment (horlah) [inline web/ui/dashboard/src/app/private/components/create-subscription-filter/create-subscription-filter.component.ts:45]: `response.data` does not return a boolean, use `!!response.data` instead.

File: web/ui/dashboard/src/app/private/components/create-subscription-filter/create-subscription-filter.component.ts [hunk 6]
```diff
@@ -74,4 +68,8 @@ export class CreateSubscriptionFilterComponent implements OnInit {
 			return false;
 		}
 	}
+
+	storeEventData() {
```
Comment (horlah) [inline web/ui/dashboard/src/app/private/components/create-subscription-filter/create-subscription-filter.component.ts:72]: Is this method ever called?

**Review by horlah — CHANGES_REQUESTED** [review]
Just these last things and we should be good

**Review by horlah — APPROVED** [review]
